src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -643,10 +672,48 @@ private boolean isAllowedApiRequest(HttpServletRequest request) {
 		if( url.matches(appExtPropertiesFile.getProperty(CANVAS_API_CROSSLIST)) && session.getAttribute(LAUNCH_TYPE).equals("lti")){
 			isAllowedRequest = isCrosslistAllowed(request, session, url);
 		}
+		
+		if( url.matches(appExtPropertiesFile.getProperty(CANVAS_API_ADD_USER)) && request.getSession().getAttribute(LAUNCH_TYPE).equals("lti")){
+			isAllowedRequest = isAddUserAllowed(request, url);
+		}
 
 		return isAllowedRequest;
 	}
 
+	private boolean isAddUserAllowed(HttpServletRequest request, String url) {
+		boolean isCallAllowed = false;
+		String sectionString = url.substring(url.indexOf("sections/")+9, url.indexOf("/enrollments"));
+		String enrollmentTypeFromRequest = url.substring(url.indexOf("enrollment[type]=")+17, url.length());
+		M_log.debug("SECTION_STRING: " + sectionString);
+		M_log.debug("ENROLLMENT_TYPE: " + enrollmentTypeFromRequest);
+		
+		HashMap<Integer, String> enrollmentsFound = (HashMap<Integer, String>) request.getSession().getAttribute("enrollments");
+		M_log.debug("SESSION ENROLLMENTS FOUND: " + request.getSession().getAttribute("enrollments"));
+		
+		String enrollmentTypeFromSession = enrollmentsFound.get(new Integer(sectionString));
+		M_log.debug("ENROLLMENT FOUND: " + enrollmentTypeFromSession);
+		
+		if(enrollmentTypeFromSession == null){
+			return isCallAllowed;
+		}
+		
+		int enrollmentValueFromRequest = enrollmentsMap.get(enrollmentTypeFromRequest);
+		int enrollmentValueFromSession = enrollmentsMap.get(enrollmentTypeFromSession);
+		
+		M_log.debug("ENROLLMENT TYPE VALUE: " + enrollmentValueFromRequest);
+		M_log.debug("ENROLLMENT FOUND VALUE: " + enrollmentValueFromSession);
+		
+		if(enrollmentValueFromSession >= enrollmentValueFromRequest){
```

Comment:
Add comment explaining why this makes sense.  If you don't someone will have to spend time tracking that story down later,